src/hal/src/command/mod.rs

```diff
@@ -155,11 +155,20 @@ impl<B: Backend, C, S: Shot, L: Level> CommandBuffer<B, C, S, L> {
 
     /// Finish recording commands to the command buffers.
     ///
-    /// The command pool must be reset to able to re-record commands.
+    /// The command buffer must be reset to able to re-record commands.
     pub unsafe fn finish(&mut self) {
         self.raw.finish();
     }
 
+    /// Empties the command buffer, optionally releasing all resources from the
+    /// commands that have been submitted.
+    ///
+    /// The command pool must have been created with the RESET_INDIVIDUAL flag
+    /// to be able to reset individual buffers.
+    pub unsafe fn reset(&mut self, release_resources: bool) {
```

Comment:
I don't think `reset` makes much sense/use for single-shot buffers. Let's move it under the impls for `Multishot` only?

Reply:
Sure, that sounds reasonable. I've pushed some changes. Is this better?

Reply:
Single shot buffers are not "one time submit and delete". Rather they should be reset before used again.